packages/client/internals/MonacoOutput.vue

```diff
@@ -0,0 +1,143 @@
+<script setup lang="ts">
+import { debounce } from '@antfu/utils'
+import { useVModel } from '@vueuse/core'
+import { computed, ref, watch } from 'vue'
+import { isDark } from '../logic/dark'
+import { isPrintMode } from '../logic/nav'
+import { runJavaScript } from '../logic/runCode'
+import { useSlideContext } from '../context'
+import IconButton from './IconButton.vue'
+
+const props = defineProps<{
+  modelValue: string
+  lang: string
+  autorun?: boolean | 'once'
+  height?: string
+}>()
+const emit = defineEmits(['update:modelValue'])
+const code = useVModel(props, 'modelValue', emit)
+
+const { $renderContext } = useSlideContext()
+const disabled = computed(() => !['slide', 'presenter'].includes($renderContext.value))
+
+const autorun = isPrintMode.value ? 'once' : props.autorun
+const output = ref(autorun ? '_running' : '_empty')
+
+let shikiModule: typeof import('#slidev/shiki') | undefined
+let tsModule: typeof import('typescript') | undefined
+
+const run = debounce(200, async () => {
+  if (disabled.value)
+    return
+
+  const setAsRunning = setTimeout(() => {
+    output.value = '_running'
+  }, 500)
+
+  const { shiki, themes } = (shikiModule ??= await import('#slidev/shiki'))
+  const highlighter = await shiki
+  const highlight = (code: string) => highlighter.codeToHtml(code, {
+    lang: 'javascript',
+    theme: typeof themes === 'string'
+      ? themes
+      : isDark.value
+        ? themes.dark || 'vitesse-dark'
+        : themes.light || 'vitesse-light',
+  })
+
+  const { transpile } = (tsModule ??= await import('typescript'))
+  const js = props.lang === 'typescript'
+    ? transpile(code.value, {
+      module: tsModule.ModuleKind.ESNext,
+    })
+    : code.value
+
+  output.value = (await runJavaScript(js)).flatMap(
+    ([type, content]) => [
+      `<div class="output-line">`,
+      `<span class="decorator">[</span>`,
+      `<span class="log-type ${type}">${type}</span>`,
+      `<span class="decorator">]:&nbsp;</span>`,
+      `<div class="select-text">`,
+      content.map(highlight).join('<span class="decorator">, </span>'),
+      `</div>`,
+      `</div>`,
+    ],
+  ).join('')
+
```

Comment:
Would be nice to render Errors as well, in some cases we might want to present certain errors in the slides.